### PR TITLE
feat(ui): preserve particle canvas across page navigation

### PR DESCRIPTION
**Pull request by willfindlay**
## Summary

- Add client-side SPA-like navigation (`static/js/navigation.js`) that intercepts internal link clicks, fetches the target page HTML, and swaps only the `<main>` content — keeping the particle canvas and its RAF animation loop running uninterrupted
- Fade-out/fade-in transition (150ms) on `<main>` runs in parallel with the fetch for smooth visual continuity
- Gracefully handles edge cases: external links, modifier-key clicks, non-HTML resources, browser back/forward, rapid clicks (AbortController), and network errors all fall back to normal navigation

## Test plan

- [x] `go test ./...` passes (no Go changes)
- [x] Click nav links — canvas persists, content swaps, title updates, nav underline moves
- [x] Click blog/project links — content swaps correctly
- [x] Browser back/forward — previous content restored, canvas persists
- [x] Ctrl+click / Cmd+click — opens new tab normally
- [x] External links (GitHub, LinkedIn) — normal navigation
- [x] Direct URL entry / refresh — full page load works normally
- [x] Scroll resets to top on each navigation

### CODE REVIEW

File: static/js/navigation.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+(() => {
+  const SKIP_EXTENSIONS = /\.(xml|pdf|svg|png|jpg|jpeg|gif|zip|tar|gz|json)$/i;
+  const TRANSITION_MS = 150;
+
+  let abortController = null;
+
+  history.scrollRestoration = "manual";
```
Comment (willfindlay) [inline static/js/navigation.js]: PR #8 (`feat/wf-particle-formation`) captures `canvas.dataset.formation` once at script init. When both PRs merge, SPA-navigating away from the homepage and back will not re-trigger the WF formation animation, because the attribute was read at page load and the canvas is preserved across navigations. Consider a coordination mechanism (e.g. a custom event the navigation emits after content swap) so particle features can respond to route changes.

File: static/js/navigation.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      if (pathname === href || pathname.startsWith(href + "/")) {
+        link.classList.add("active");
+      } else {
+        link.classList.remove("active");
+      }
+    }
+  }
+
+  function fadeOut(main) {
+    main.style.opacity = "0";
```
[the remark this answers is not in the document]
Comment (willfindlay) [reply static/js/navigation.js]: Fixed. Added a `reducedMotion` const using the same `window.matchMedia("(prefers-reduced-motion: reduce)")` pattern from `particles.js`. The fade-out resolves immediately via `Promise.resolve()` and `fadeIn` is skipped entirely when reduced motion is preferred.
